src/states/joining_peer.rs

```diff
@@ -310,8 +310,15 @@ impl Base for JoiningPeer {
         }
 
         let msg = msg.into_signed_routing_message();
+
+        trace!(
+            "{} - Handle signed message: {:?}",
+            self,
+            msg.routing_message()
+        );
+
         if self.in_location(&msg.routing_message().dst) {
-            self.check_signed_message_integrity(&msg)?;
+            // TODO: verify the message
```

Comment:
still need to left this TODO over?

Reply:
We currently can't properly verify a message when we are `JoiningPeer`. This needs to be addressed separately (it's one of the remaining points in https://github.com/maidsafe/routing/issues/1835).